configuration/pih/scripts/global/htmlFormFunctions.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Adds javascript functions for use within html forms
+ * Requires jQuery and moment js
+ */
+class HtmlFormFunctions {
+
+    constructor(pihemr, patientUuid, encounterUuid, encounterDateStr) {
+        this.pihemr = pihemr;
+        this.jq = pihemr.getJq();
+        this.patientUuid = patientUuid;
+        this.encounterUuid = encounterUuid;
+        this.encounterDateStr = encounterDateStr;
+    }
+
+    getPatientUuid() {
+        return this.patientUuid;
+    }
+
+    getEncounterUuid() {
+        return this.encounterUuid;
+    }
+
+    getEncounterDateYmd() {
```

Comment:
Actually, referring to the above, this may fail in some cases (though admittedly this is not not well documented).  The "encounterDate" element only had an id of "encounterDate" if no specific id parameter is given:

https://github.com/openmrs/openmrs-module-htmlformentry/blob/master/api/src/main/java/org/openmrs/module/htmlformentry/element/EncounterDetailSubmissionElement.java#L563

And specifically, the property accessor stuff theoretically would be updated if we modified the structure of the encounter date widget behind the scenes.

Reply:
Yes, it may fail in some cases, which is why the function is in the pihemr config and we can use it in places where we know it will work (eg. have standards for what ids we place on our form elements).  I see no suitable features that exist in HFE already that will meet the same needs that these provide.